lib/Analysis/MulDepthAnalysis/MulDepthAnalysis.h

```diff
@@ -0,0 +1,81 @@
+#ifndef LIB_ANALYSIS_MULDEPTHANALYSIS_MULDEPTHANALYSIS_H_
+#define LIB_ANALYSIS_MULDEPTHANALYSIS_MULDEPTHANALYSIS_H_
+
+#include "mlir/include/mlir/Analysis/DataFlow/SparseAnalysis.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/Diagnostics.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/Operation.h"    // from @llvm-project
+#include "mlir/include/mlir/IR/Value.h"        // from @llvm-project
+
+namespace mlir {
+namespace heir {
+namespace mul_depth_analysis {
+
+class MulDepth {
+ public:
+  MulDepth() : value(std::nullopt) {}
+  MulDepth(int64_t value) : value(value) {}
```

Comment:
To avoid unexpected conversion casts, prepend this constructor with `explicit`
```suggestion
  explicit MulDepth(int64_t value) : value(value) {}
```